awx/api/views/__init__.py

```
@@ -1622,23 +1647,56 @@ def post(self, request, *args, **kwargs):
         except requests.exceptions.HTTPError:
             message = """Test operation is not supported for credential type {}.
                 This endpoint only supports credentials that connect to
                 external secret management systems such as CyberArk, HashiCorp
-                Vault, or cloud-based secret managers.""".format(obj.credential_type.kind)
-            return Response({'detail': message}, status=status.HTTP_400_BAD_REQUEST)
+                Vault, or cloud-based secret managers.""".format(
+                obj.credential_type.kind
+            )
+            response_body['details']['error_message'] = message
+            return Response(response_body, status=status.HTTP_400_BAD_REQUEST)
         except Exception as exc:
```

Ruff (0.15.7)

[warning] Do not catch blind exception: `Exception`

(BLE001)